### PR TITLE
DRIVERS-2789 Convert Transactions Convenient API Spec to Markdown

### PR DESCRIPTION
**Pull request by blink1073**
Please complete the following before merging:

- [x] Update changelog.
- [ ] Make sure there are generated JSON files from the YAML test files.
- [ ] Test changes in at least one language driver.
- [ ] Test these changes against all server versions and topologies (including standalone, replica set, sharded
  clusters, and serverless).

### CODE REVIEW

File: source/transactions-convenient-api/transactions-convenient-api.md [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+### It is assumed that the callback will not start a new transaction on the ClientSession
+
+Under normal circumstances, the callback should not commit the transaction nor should it start a new transaction. The
+`withTransaction` method will inspect the ClientSession's transaction state after the callback returns and take the most
+sensible course of action; however, it will not detect whether the callback has started a new transaction.
+
+### The callback function may be executed multiple times
+
+The implementation of withTransaction is based on the original examples for
+[Retry Transactions and Commit Operation](https://www.mongodb.com/docs/manual/core/transactions/#retry-transaction-and-commit-operation)
```
Comment (jmikola) [inline source/transactions-convenient-api/transactions-convenient-api.md:294]: This is a dead anchor. You could conceivably link to https://www.mongodb.com/docs/manual/core/transactions-in-applications/#core-api instead, but at this point I think this entire section could just be reduced to:

> The callback may be executed any number of times. Drivers are free to encourage their
users to design idempotent callbacks.

Ignore this suggestion if you'd rather not make the change in this PR.

File: source/transactions-convenient-api/transactions-convenient-api.md [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+### The commit is not retried after a MaxTimeMSExpired error
+
+This specification intentionally chooses not to retry commit operations after a MaxTimeMSExpired error as doing so would
+exceed the user's original intention for `maxTimeMS`.
+
+### The transaction and commit may be retried any number of times within a timeout period
+
+The implementation of withTransaction is based on the original examples for
+[Retry Transactions and Commit Operation](https://www.mongodb.com/docs/manual/core/transactions/#retry-transaction-and-commit-operation)
```
Comment (jmikola) [inline source/transactions-convenient-api/transactions-convenient-api.md:329]: Same feedback as I left for "The callback function may be executed multiple times" above. This is a dead anchor.

This paragraph could be reduced to:

> The transaction and commit may be continually retried as long as the error label
indicates that retrying is possible.

File: source/transactions-convenient-api/tests/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Convenient API for Transactions Tests
+
+______________________________________________________________________
+
+## Introduction
+
+The YAML and JSON files in this directory are platform-independent tests meant to exercise a driver's implementation of
+the Convenient API for Transactions spec. These tests utilize the
+[Unified Test Format](../../unified-test-format/unified-test-format.rst).
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/transactions-convenient-api/tests/README.md:9]: Yeah, I'm going to do a full audit of links as a follow up task.

File: source/transactions-convenient-api/tests/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+Write a callback that returns a custom value (e.g. boolean, string, object). Execute this callback using
+`withTransaction` and assert that the callback's return value is propagated to the caller of `withTransaction`.
+
+### Retry Timeout is Enforced
+
+Drivers should test that `withTransaction` enforces a non-configurable timeout before retrying both commits and entire
+transactions. Specifically, three cases should be checked:
+
+> - If the callback raises an error with the TransientTransactionError label and the retry timeout has been exceeded,
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/transactions-convenient-api/tests/README.md:32]: Done

File: source/transactions-convenient-api/transactions-convenient-api.md [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+
+## Changelog
+
+- 2024-09-06: Migrated from reStructuredText to Markdown.
+
+- 2023-11-22: Document error handling inside the callback.
+
+- 2022-10-05: Remove spec front matter and reformat changelog.
+
+- 2022-01-19: withTransaction applies timeouts per the client-side operations\
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/transactions-convenient-api/transactions-convenient-api.md:379]: Removed

File: source/transactions-convenient-api/transactions-convenient-api.md [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+implementations may be found via [DRIVERS-556](https://jira.mongodb.org/browse/DRIVERS-556).
+
+## Security Implication
+
+Applications that use transaction guarantees to enforce security rules will benefit from a less error-prone API. Adding
+a helper method to execute a user-defined function within a transaction has few security implications, as it only
+provides an implementation of a technique already described in the MongoDB 4.0 documentation
+([DRIVERS-488](https://jira.mongodb.org/browse/DRIVERS-488)).
+
+## Future Work
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/transactions-convenient-api/transactions-convenient-api.md:364]: Removed

File: source/transactions-convenient-api/transactions-convenient-api.md [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+Driver object representing a client session, as defined in the [Driver Session](../sessions/driver-sessions.md)
+specification. The name of this object MAY vary across drivers.
+
+**MongoClient**
+
+The root object of a driver's API. The name of this object MAY vary across drivers.
+
+<div id="TransactionOptions">
+
+TransactionOptions\
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/transactions-convenient-api/transactions-convenient-api.md:42]: Fixed

File: source/transactions-convenient-api/transactions-convenient-api.md [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+  `{ok:1, writeConcernError: {code: 50, codeName: "MaxTimeMSExpired"}}`.
+
+A previous design for `withTransaction` retried for all of these errors *except* for write concern timeouts, so as not
+to exceed the user's original intention for `wtimeout`. The current design of this specification no longer excludes
+write concern timeouts, and simply retries `commitTransaction` within its timeout period for all errors bearing the
+"UnknownTransactionCommitResult" label.
+
+This change was made in light of the forthcoming Client-side Operations Timeout specification (see:
+[Future Work](#future-work)), which we expect will allow the current 120-second timeout for `withTransaction` to be
+customized and also obviate the need for users to specify `wtimeout`.
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/transactions-convenient-api/transactions-convenient-api.md:319]: Removed

File: source/transactions-convenient-api/transactions-convenient-api.md (hunk 1 above)
Comment (blink1073) [reply source/transactions-convenient-api/transactions-convenient-api.md:294]: Done

File: source/transactions-convenient-api/transactions-convenient-api.md (hunk 2 above)
Comment (blink1073) [reply source/transactions-convenient-api/transactions-convenient-api.md:325]: Done

**Review by dariakp — COMMENTED** [review]
This doesn't look like something that needs 3 reviewers - Jeremy's review looks pretty comprehensive, but let me know if you think anything actually needs an extra pair of eyes.

**Review by jmikola — APPROVED** [review]

**Pull request merged**